### PR TITLE
feat(flutter_shopping_cart): add delete functionality 

### PR DESCRIPTION
**Pull request by elianmortega**
## Status

**READY**

## Breaking Changes

NO

## Description

- Added delete functionality to `CartBloc`.

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: examples/flutter_shopping_cart/lib/cart/bloc/cart_bloc.dart [hunk 1]
```diff
@@ -48,4 +50,26 @@ class CartBloc extends Bloc<CartEvent, CartState> {
...
+    CartState state,
+  ) async* {
+    if (state is CartLoaded) {
+      try {
+        shoppingRepository.removeItemFromCart(event.item);
+
+        yield CartLoaded(
+          cart: Cart(
+            items: [
+              for (final item in state.cart.items)
```
Comment (elianmortega) [inline examples/flutter_shopping_cart/lib/cart/bloc/cart_bloc.dart:65]: Kudos to [@marcossevilla](https://github.com/marcossevilla) for these two lines

**Comment by marcossevilla** [comment]
Hey @elianmortega, left a PR to migrate your functionality to bloc 7.2.0 so it can be merged to the current state of `master`.

When you accept it, this PR should be good to go. Thanks for your contribution! 💯

**Comment by elianmortega** [comment]
He @marcossevilla, I'll be doing that during the weekend 👍

**Comment by felangel** [comment]
> He @marcossevilla, I'll be doing that during the weekend 👍

Do you still have time to do this? If not, I'm happy to try to address the feedback for you.

**Review by marcossevilla — APPROVED** [review]
LGTM ⚡️

File: examples/flutter_shopping_cart/test/cart/view/cart_page_test.dart [hunk 2]
```diff
@@ -19,6 +19,8 @@ void main() {
     Item(3, 'item #3'),
   ];
 
+  final mockItemToRemove = mockItems.last;
```
Comment (felangel) [inline examples/flutter_shopping_cart/test/cart/view/cart_page_test.dart:22]: Can we move this variable into the test where it's being used?

File: examples/flutter_shopping_cart/lib/cart/bloc/cart_bloc.dart [hunk 3]
```diff
@@ -37,4 +38,25 @@ class CartBloc extends Bloc<CartEvent, CartState> {
...
+    if (state is CartLoaded) {
+      try {
+        shoppingRepository.removeItemFromCart(event.item);
+        emit(
+          CartLoaded(
+            cart: Cart(
+              items: [
+                for (final item in state.cart.items)
+                  if (item != event.item) item
+              ],
```
[the remark this answers is not in the document]
Comment (elianmortega) [reply examples/flutter_shopping_cart/lib/cart/bloc/cart_bloc.dart]: Solved in 238e56e

**Review by marcossevilla — DISMISSED** [review]

**Review by felangel — APPROVED** [review]
LGTM thanks so much for the contribution! 🎉 🚀

**Pull request merged**